gitbugactions/test_executor.py

```diff
@@ -14,14 +22,54 @@ def __init__(
         default_actions: GitHubActions,
         runner_image: str = "gitbugactions:latest",
     ):
+        TestExecutor.__schedule_cleanup(runner_image)
         self.act_cache_dir = act_cache_dir
         self.repo_clone = repo_clone
         self.runner_image = runner_image
         self.language = language
-        # Note: these default actions may have different configuration options such as paths, runners, etc.
+        # Note: these default actions may have different configuration options
+        # such as paths, runners, etc.
         self.default_actions = default_actions
         self.first_commit = repo_clone.revparse_single("HEAD")
 
+    @staticmethod
+    def __schedule_cleanup(runner_image):
+        with TestExecutor.__CLEANUP_LOCK:
+            if (
+                runner_image in TestExecutor.__CLEANUP
+                or not TestExecutor.__CLEANUP_ENABLED
+            ):
+                return
+
+            docker = DockerClient.getInstance()
+
+            def cleanup():
+                for container in docker.containers.list(
+                    all=True, filters={"ancestor": runner_image, "status": "exited"}
+                ):
+                    container.remove()
```

Comment:
```suggestion
                    container.remove(force=True)
```

Maybe forcing here is even better?

Reply:
Yap that sounds like a good idea